### PR TITLE
Jnasim/dpc 2815

### PR DESCRIPTION
**Pull request by knasim**
## Change Details

* Added new header to DPC code per BFD README.

## Security Implications

- [ ] new software dependencies

- [ ] security controls or supporting software altered

- [ ] new data stored or transmitted

- [ ] security checklist is completed for this change

- [ ] requires more information or team discussion to evaluate security implications

- [ ] PHI/PII is affected by this change

### CODE REVIEW

File: dpc-aggregation/src/main/java/gov/cms/dpc/aggregation/engine/JobBatchProcessor.java [hunk 1]
```diff
@@ -130,26 +128,11 @@ private Flowable<Resource> fetchResource(JobQueueBatch job, String patientID, DP
                 resourceType,
                 since,
                 job.getTransactionTime());
-        return fetcher.fetchResources(patientID, buildHeaders(job))
+        //return fetcher.fetchResources(patientID, buildHeaders(job))
```
Comment (mftcodes) [inline dpc-aggregation/src/main/java/gov/cms/dpc/aggregation/engine/JobBatchProcessor.java:131]: Can we remove this comment please?

File: dpc-aggregation/src/main/java/gov/cms/dpc/aggregation/engine/JobHeaders.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        headers.put(HttpHeaders.X_FORWARDED_FOR, job.getRequestingIP());
+        headers.put(Constants.BlueButton.ORIGINAL_QUERY_ID_HEADER, job.getJobID().toString());  //test this
+        if (job.isBulk()) {
+            headers.put(Constants.BULK_JOB_ID_HEADER, job.getJobID().toString());
+            headers.put(Constants.BULK_CLIENT_ID_HEADER, job.getProviderNPI()); //test this
+            headers.put(Constants.BlueButton.BULK_CLIENTNAME_HEADER,Constants.BlueButton.APPLICATION_NAME_DESC); //test this
+        } else {
+            headers.put(Constants.DPC_CLIENT_ID_HEADER, job.getProviderNPI()); //test this
+            headers.put(Constants.BlueButton.APPLICATION_NAME_HEADER,Constants.BlueButton.APPLICATION_NAME_DESC); //test this
+            headers.put(Constants.BlueButton.ORIGINAL_QUERY_TIME_STAMP_HEADER,job.getTransactionTime().toString()); //test this
```
Comment (mftcodes) [inline dpc-aggregation/src/main/java/gov/cms/dpc/aggregation/engine/JobHeaders.java:36]: Do we need the `//test this` comments on these lines?

File: dpc-aggregation/src/main/java/gov/cms/dpc/aggregation/engine/JobHeaders.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        }
+        return headers;
+    }
+}
```
Comment (mftcodes) [inline dpc-aggregation/src/main/java/gov/cms/dpc/aggregation/engine/JobHeaders.java:40]: I like the use of class here!

**Review by mftcodes — CHANGES_REQUESTED** [review]
Check comments. I think one comment we can remove, and the others I wonder if they were left for yourself as a reminder or testing?

File: dpc-aggregation/src/main/java/gov/cms/dpc/aggregation/engine/JobBatchProcessor.java (hunk 1 above)
Comment (lgarceau768) [reply dpc-aggregation/src/main/java/gov/cms/dpc/aggregation/engine/JobBatchProcessor.java:131]: I second this change

File: dpc-aggregation/src/main/java/gov/cms/dpc/aggregation/engine/JobHeaders.java (hunk 2 above)
Comment (lgarceau768) [reply dpc-aggregation/src/main/java/gov/cms/dpc/aggregation/engine/JobHeaders.java:36]: Yes please remove the `//test this`

**Review by lgarceau768 — CHANGES_REQUESTED** [review]
Some minute changes

File: dpc-aggregation/src/main/java/gov/cms/dpc/aggregation/engine/JobBatchProcessor.java (hunk 1 above)
Comment (knasim) [reply dpc-aggregation/src/main/java/gov/cms/dpc/aggregation/engine/JobBatchProcessor.java:131]: removed.

File: dpc-aggregation/src/main/java/gov/cms/dpc/aggregation/engine/JobHeaders.java (hunk 2 above)
Comment (knasim) [reply dpc-aggregation/src/main/java/gov/cms/dpc/aggregation/engine/JobHeaders.java:36]: they are removed now - added those as reminders to myself.

**Review by lgarceau768 — APPROVED** [review]

**Comment by knasim** [comment]
Dismissed review in order to allow secondary reviewer to complete the merge.  Primary reviewer is OOO till next year and this blocks secondary reviewer from merging.

**Pull request merged**